### PR TITLE
build: integrate null type in Class Definition

### PR DESCRIPTION
**Pull request by summer-ji-eng**
A class cannot extends NullType because it is an interface.

But a class can implements an NullType, so I put two cases in `classdefinitiontest`
Here is an example to implement NullType, where it is imported from `javax.lang.model.type.NullType`
```
class Summer implements NullType {

  @Override
  public TypeKind getKind() {
    return null;
  }

  @Override
  public <R, P> R accept(TypeVisitor<R, P> v, P p) {
    return null;
  }

  @Override
  public List<? extends AnnotationMirror> getAnnotationMirrors() {
    return null;
  }

  @Override
  public <A extends Annotation> A getAnnotation(Class<A> annotationType) {
    return null;
  }

  @Override
  public <A extends Annotation> A[] getAnnotationsByType(Class<A> annotationType) {
    return null;
  }
}
```

### CODE REVIEW

**Comment by miraleung** [comment]
Not related to this PR, but this reminded me that we will need to ensure that `NullType` does not get imported when ImportWriterVisitor visits NullObjectValue. Please feel free to add this work to our tracker as well.

Also, if we end up having to generate code that extends or implements the actual `NullType` class, we can refactor this to use our own `ast.NullType` class as the underlying representative class.

**Comment by summer-ji-eng** [comment]
> Since we likely won't be generating code that `implements NullType`, I think it's OK to prevent that case as well, so we can ensure no one does `implements null`.

Cool. Thanks comment out.

**Comment by summer-ji-eng** [comment]
> Not related to this PR, but this reminded me that we will need to ensure that `NullType` does not get imported when ImportWriterVisitor visits NullObjectValue. Please feel free to add this work to our tracker as well.
> 
> Also, if we end up having to generate code that extends or implements the actual `NullType` class, we can refactor this to use our own `ast.NullType` class as the underlying representative class.

**Comment by summer-ji-eng** [comment]
> Not related to this PR, but this reminded me that we will need to ensure that `NullType` does not get imported when ImportWriterVisitor visits NullObjectValue. Please feel free to add this work to our tracker as well.

Cool. Added in our tracksheet.

File: src/test/java/com/google/api/generator/engine/ast/ClassDefinitionTest.java [hunk 1]
```diff
@@ -107,6 +108,32 @@ public void validClassDefinition_statementsAndMethods() {
...
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          ClassDefinition.builder()
+              .setPackageString("com.google.example.library.v1.stub")
+              .setName("LibraryServiceStub")
+              .setScope(ScopeNode.PUBLIC)
+              .setImplementsTypes(
+                  Arrays.asList(
+                      TypeNode.withReference(ConcreteReference.withClazz(NullType.class))))
```
Comment (miraleung) [inline src/test/java/com/google/api/generator/engine/ast/ClassDefinitionTest.java:122]: Did you mean `TypeNode.NULL`?

Comment (summer-ji-eng) [reply src/test/java/com/google/api/generator/engine/ast/ClassDefinitionTest.java:122]: I try to cover two cases here. But I guess we do not need cover this one right now. I will remove it.

**Pull request merged**